lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -62,18 +83,144 @@ struct LowerClassesPass : public LowerClassesBase<LowerClassesPass> {
   void updateInstances(Operation *op);
 
   // Convert to OM ops and types in Classes or Modules.
-  LogicalResult dialectConversion(Operation *op);
+  LogicalResult dialectConversion(Operation *op,
+                                  const PathInfoTable &pathInfoTable);
 };
 
 } // namespace
 
+/// This pass removes the OMIR tracker annotations from operations, and ensures
+/// that each thing that was targeted has a hierarchical path targeting it. /
+/// It builds a table which maps the OMIR trackker annotation IDs to the
+/// corresponding hierarchical paths. We use this table to convert FIRRTL path
+/// ops to OM. FIRRTL paths refer to their target using a target ID, while OM
+/// paths refer to their target using Hierarchical paths.
+static LogicalResult lowerPaths(CircuitOp circuit,
+                                PathInfoTable &pathInfoTable) {
+  SymbolTable symbolTable(circuit);
+  hw::InnerSymbolNamespaceCollection namespaces;
+  HierPathCache cache(circuit, symbolTable);
+  InstanceGraph instanceGraph(circuit);
+  auto *context = circuit.getContext();
+
+  auto processPathTrackers = [&](AnnoTarget target) -> LogicalResult {
+    auto error = false;
+    auto annotations = target.getAnnotations();
+    auto *op = target.getOp();
+    FModuleLike module;
+    annotations.removeAnnotations([&](Annotation anno) {
+      // If there has been an error, just skip this annotation.
+      if (error)
+        return false;
+
+      // We are looking for OMIR tracker annotations.
+      if (!anno.isClass("circt.tracker"))
```

Comment:
Can this string constant live somewhere? Maybe with the other annotation class strings in that header?

Reply:
will do this in a follow up